### ISSUE TITLE
Set max-age for blob when storing file to Azure blob storage

### ISSUE THREAD

**Issue by FanrayMedia** [FanrayMedia/Fanray#122]
# Intro

As part of [Google PageSpeed Tools > Insight](https://developers.google.com/speed/pagespeed/insights/), it recommends to [Leverage browser caching](https://developers.google.com/speed/docs/insights/LeverageBrowserCaching), it gives the following warning if [cache-control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) is not used

> Setting an expiry date or a maximum age in the HTTP headers for static resources instructs the browser to load previously downloaded resources from local disk rather than over the network.

For example my image is shown up on the warning list

> https://fanray.blob.core.windows.net/…g/1/2017/11/01-custom-domain-thumb-1.png (expiration not specified)

## Asp.net Core Response Cachine

This reading is necessary from Microsoft doc [HTTP-based response caching](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response?view=aspnetcore-2.0#http-based-response-caching)

## Blob Storage

The SO question on [Add Cache-Control and Expires headers to Azure Storage Blobs
](https://stackoverflow.com/q/4507657/32240) shows more info.

Basically add this code to AzureBlobStorageProvider

`blob.Properties.CacheControl = "public, max-age=31536000"; // 1 yr`

There is a `SetProperties()` that does not seem to need.

## Azure Storage Explorer

You can verify locally using ASE, open the properties on a image to see all its properties.

## Further Reading

[Using ImageSharp to resize images in ASP.NET Core - Part 4: saving to disk](https://andrewlock.net/using-imagesharp-to-resize-images-in-asp-net-core-part-4-saving-to-disk/)
